runhouse/resources/secrets/provider_secrets/provider_secret.py

```diff
@@ -155,9 +155,14 @@ def to(
         Example:
             >>> secret.to(my_cluster, path=secret.path)
         """
+        from runhouse import Env
+
         system = _get_cluster_from(system)
         path = path or self.path
 
+        if not env or (env and isinstance(env, Env) and not env.name):
```

Comment:
secrets should be a general resource that works across envs/clusters, rather than have a specific env attribute for example. otherwise can end up with duplicate resources repr same secret but in a diff env